Adding Landing Page with functionaality via issue #6

pjconnolly12 opened the pull request:
#6 
Here is the functional Landing Page component. 

In order to get the location to work you will need to create an API Key with Open Cage Data. It's very easy to do and we get 2,500 calls per day.

https://opencagedata.com/api

Create a .env file in the root of the project
Add `REACT_APP_GEO_CODER_API_KEY= "API Key Here"`

Features:
Search Input will automatically update with users City and State
Click on different features to narrow down search

As discussed before, Design aspect will be completed separately

pjconnolly12 committed:
Adding Landing Page with functionaality via issue #6

Signed-off-by: unknown <pjconnolly12@gmail.com>

andreareed commented on client/src/components/landingpage.js:
```diff
@@ -0,0 +1,60 @@
+			.then((response) => {
+				let city = response.data.results[0].components.town
+				let state = response.data.results[0].components.state_code
+				setPlaceholder(`${city}, ${state}`)
```
I'm not familiar with this api - What happens here if we get no results back?  If it still returns a 200, this will throw an error.

pjconnolly12 replied:
currently I have it set so that if an error happens it just displays the default placeholder. Should we add an error saying their location could not be found?

TheHuns replied:
Yeah i dont think that we need to let them know that the geolocation part didnt find them.  Just a message if the actual search fails to return anything. Which I assume we'll handle in redux or the apollo-graphql tools.

andreareed commented on client/src/components/landingpage.js:
```diff
@@ -0,0 +1,60 @@
+				let state = response.data.results[0].components.state_code
+				setPlaceholder(`${city}, ${state}`)
+			}, (error) => {
+				console.log(error)
```
We should probably do something to handle any errors.  Maybe set it in state and display something below the input?

pjconnolly12 replied:
This is the same idea as above, we could display a message saying their location could not be found.

andreareed commented on client/src/components/landingpage.js:
```diff
@@ -0,0 +1,60 @@
+			})
+	}
+
+	function handleIconClick(e) {
```
No arrow function or binding?  Do hooks negate the need for it?  Neat 👍

pjconnolly12 replied:
yes hooks negate the need for those because you don't use class components, you use functional components. I'm fairly new to hooks still but I am liking them a lot as I think the code is much cleaner. 

We can also refactor this once we get some sort of state management setup

andreareed reviewed (changes requested):
This looks great, thanks for being the brave soul to go first! 😉

A couple of small things I'd like to see changed - 
1. Error handling for the location search
2. Update the README with information about the getting the api key so it's in a good place without having to track down this PR
3.  Replace ☝️ with the issue number so it will link your PR to the issue in case we ever _do_ have to track it down for some reason.  You just have to do # before the issue number and GH does the rest, like `#6`

Nice work!

TheHuns reviewed (commented):
Looking good to me, I'm sure we'll need to make some changes once we get the api and redux going. I would say just get the update to the README pushed and we can probably move on.  && number 3 on @andreareed comment above

pjconnolly12 committed:
added error handling and updated README

pjconnolly12 commented:
I updated the README and I also added some of the error handling, still more to be added though. I also fixed the re-render issue so it won't fetch the API every time you do something.

TheHuns committed:
Update README.md
